### ISSUE TITLE
Game does not translate under latest patch

### ISSUE THREAD

**Issue by john-yick** [thomas-barthelemy/QQ-LoL-Localizer#7]
Looking at the files it does look like the english has been injected into the correct files, but it seems like the game now pulls the text from another source?

**Comment by thomas-barthelemy** [comment]
I will check that over the weekend, is that happening on the Air client, in game or both?

**Comment by john-yick** [comment]
It seems like it is just the game, the Air client seems to be translated correctly.

**Comment by TheWalkingDeaf** [comment]
same here. air client translated, also in game is translated, but i can't read the chinese text now.

**Comment by FredRiv974** [comment]
to confirme: air client translated, In Game is not

**Comment by TheWalkingDeaf** [comment]
in game is translated, i see everything in english, the problem is that i can't read chinese in chat, or every chinese character in game

**Comment by john-yick** [comment]
So this is what I am seeing with the latest patch. I think it has to do with the way the game was updated maybe, but I am unsure. Everything looks like it has translated fine, double checked the fontconfigzh_cn file to make sure.

![image](https://cloud.githubusercontent.com/assets/17866270/17461562/b0c09ed8-5cc3-11e6-982f-1d06812f7fde.png)

![image](https://cloud.githubusercontent.com/assets/17866270/17461572/221714ea-5cc4-11e6-9bef-0eeb193f43bc.png)

Thanks,
John

**Comment by TheWalkingDeaf** [comment]
it's different from what i get:
![lol1](https://cloud.githubusercontent.com/assets/20873944/17461683/8591637e-5cc7-11e6-8699-9beba78abdf7.jpg)
![lol2](https://cloud.githubusercontent.com/assets/20873944/17461685/8a48db90-5cc7-11e6-8d59-592227f1f71a.jpg)
![lol3](https://cloud.githubusercontent.com/assets/20873944/17461686/8b53fc90-5cc7-11e6-91f8-02cfc92f82ac.jpg)

hope it can be of help,
thank you

**Comment by thomas-barthelemy** [comment]
All that strongly point to a change in the font used that I need to update. As I'm not in China anymore I had to get latest version of the game over the weekend, I should be able to give it a fix tonight hopefully!

The faulty piece would be at `Game\Data\Menu\fontconfig_zh_CN.txt` and `Game\Data\Menu\fontconfig_en_US.txt`

The fix action on that file backup the zh_CN, copy the en_US over it and replace in that file:
 - `fonts_latin.swf` by `fonts_ch.swf`
 - `Gill Sans MT Pro Medium` by `FZLanTingHei-L-GBK`

My bet is that in the original zh_CN either `fonts_ch.swf` or `FZLanTingHei-L-GBK` have been replaced by something else.

**Comment by thomas-barthelemy** [comment]
So I checked a bit and it's more annoying that expected.
Before anything, @john-yick screens shows what you should get with the latest version.
@TheWalkingDeaf shows what you might get if the `\Game\DATA\CFG\Locale.cfg` is set to `en_US` instead of `zh_CN`.

So the tricky part now: It seems the configurations aren't read from the text files anymore (so the modifications on the fontconfig_zh_CN are just not helping anymore.
I would expect them to have packaged those config into binaries that would be a pain to edit.

So no fix for now, I'll give it another look using another way, trying to switch the overall Locale.cfg to en_US and replacing the fonts directly, might help (fontmapping doesn't help either).

**Comment by TheWalkingDeaf** [comment]
@thomas-barthelemy that's what i already tried, i guess that's the reason why i get the english, differently from @john-yick .
let me know if i can help somehow =). (don't expect much... but i can be a good lab-rat if you need!).

**Comment by thomas-barthelemy** [comment]
From why I checked this morning the locale.cfg changes the file that is loaded from
`Game\Datas2.zip\DATA\FINAL\Localized\Global.xx_XX.wad.client` and those binary pack the translations. The good news is that using that we could translate the in-game client in any language now but might run into font problem for most of them.

**Workaround for getting the client in English**

Now that we have all locals packed in the zip, it's time to be smart:
we can put in the locale.cfg: `en_SG`, which will load the Singapore locale. Singapore having multiple official languages including Mandarin, it will loads the right font to support Chinese while keeping the UI in English.

@TheWalkingDeaf if you could confirm that works and no trouble happen during a game, that would help as I can only test with custom games.

**Comment by TheWalkingDeaf** [comment]
@thomas-barthelemy , i'll definetly give it a go tonight, keep you updated! thank you =)

**Comment by thomas-barthelemy** [comment]
@TheWalkingDeaf You can give a try to the v2.10 pre-release and see how that works

https://github.com/thomas-barthelemy/QQ-LoL-Localizer/releases/tag/v2.10

let me know! Thanks for the support along that fix so far to everybody anyway!

**Comment by TheWalkingDeaf** [comment]
works perfectly! i'll keep an eye on eventual bugs and let you know ;)
thank you for your work.

**Comment by TheWalkingDeaf** [comment]
Hello Thomas, 

 

Is still Marco, I contacted you not so long ago about your patcher. Right now the translation is perfect, every aprt of the game is translated well, so thanks a lot for that.

 

Unfortuantely I noticed another problem, I get random freezes of the whole PC while I’m in game. Is it possible that the patch modifies something that made the game and my pc crash? It happens only with LoL, and it started happening after I downloaded the latest version of the patch. 

 

For Now, I’m trying to reinstall the game and use it without the patch for a little bit, see if it happens again.

 

Will keep you posted anyway,

Thanks for your attention
